x-pack/plugins/data_enhanced/server/search/es_search_strategy.ts

```diff
@@ -81,19 +75,17 @@ export const enhancedEsSearchStrategyProvider = (
 
   const cancel = async (context: RequestHandlerContext, id: string) => {
     logger.info(`cancel ${id}`);
-    const method = 'DELETE';
-    const path = encodeURI(`/_async_search/${id}`);
-    await context.core.elasticsearch.legacy.client.callAsCurrentUser('transport.request', {
-      method,
-      path,
+    await context.core.elasticsearch.client.asCurrentUser.transport.request({
```

Comment:
Can we add TODOs that reference the issue that needs to be fixed in order to switch to the client API for this? Same thing below.